### DISCUSSION TITLE
Newcomer questions

### DISCUSSION THREAD

**Discussion by brupelo** [asabaylus/react-command-palette#698] [Q&A] [answered]
Hello, first of all let me tell you this project is pretty awesome, I've started working with it today and I love it :) . I've got
some simple questions, maybe I should have opened different issues for each one of my questions but I didn't want
to create too many issues so I've just created this generic thread... Please let me know if these type of questions should be asked in other place though. Here you go:

- Is it possible somehow to consider the command's category when searching? Ie. I'm using sampleCommandChrome where you can see the command's category displayed in some nice color but these ones won't be taken into consideration on the search (I assume only the name will be considered).
- Is it possible to make invisible the command palette button on the page? I just want to trigger the palette when the shortcut is pressed
- I don't see any vertical scrollbar like in Sublime/Atom/... when the number of search results is too big, would it be possible to have that somehow or it's just not implemented? In fact, even if I use `maxDisplayed={20}` attribute the number of commands shown will be less than that
- How would you associate a context object on the command functions? Ie: I'd like the commands to use some widgets references.

Thanks in advance :)

**Comment by asabaylus** [comment, marked as answer]
Glad your finding it useful.

> * Is it possible somehow to consider the command's category when searching? Ie. I'm using sampleCommandChrome where you can see the command's category displayed in some nice color but these ones won't be taken into consideration on the search (I assume only the name will be considered).
Duplicate of 

Yes, there's an example you can look at here: https://react-command-palette.js.org/?path=/story/command-palette--with-multiple-highlights. You can see this in practices in the Chrome theme example here: https://github.com/asabaylus/react-command-palette/blob/release/examples/sampleChromeCommand.js#L18-L25
```
<CommandPalette
  commands={commands}
  options={{keys: ["name", "category"]}}
  renderCommand={sampleChromeCommand}
 />
```
> * Is it possible to make invisible the command palette button on the page? I just want to trigger the palette when the shortcut is pressed

Yes but it will hurt A11Y, anything can be a trigger. 
```
<CommandPalette commands={data} trigger={<div></div>}>
```
You can also disable the trigger and manually open / close

> * I don't see any vertical scrollbar like in Sublime/Atom/... when the number of search results is too big, would it be possible to have that somehow or it's just not implemented? In fact, even if I use maxDisplayed={20} attribute the number of commands shown will be less than that

Use a larger number of items than the `suggestionsList` will accommodate, ex: `maxDisplayed=100`  then in a theme you can apply a CSS class which add `overflow: auto`:

```
.atom-suggestionsContainerOpen {
    overflow: auto;
    max-height: 315px;
}
```
There's an example here: https://codesandbox.io/s/react-command-palette-with-atom-theme-2qb4m?fontsize=14&file=/src/atomTheme.css:1441-1474

> * How would you associate a context object on the command functions? Ie: I'd like the commands to use some widgets references.

I'm not sure I understand the question, can you share a practical use case?
